core/src/main/java/com/linecorp/armeria/client/BlockingConditionalResponseAs.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.function.Predicate;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.ResponseEntity;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * Provides a way for users to add {@link ResponseAs} mappings to transform an aggregated response
+ * given that the corresponding {@link Predicate} is satisfied. Note that the conditionals are
+ * invoked in the order in which they are added.
+ *
+ * <pre>{@code
+ * RestClient.of(...)
+ *   .get("/")
+ *   .execute(
+ *     ResponseAs.<MyResponse>json(MyMessage.class, res -> res.status().isError())
+ *       .andThenJson(EmptyMessage.class, res -> res.status().isInformational())
+ *       .orElseJson(MyError.class);
+ * }</pre>
+ */
+@UnstableApi
+public final class BlockingConditionalResponseAs<V>
+        extends DefaultConditionalResponseAs<HttpResponse, AggregatedHttpResponse, ResponseEntity<V>> {
+    private static final Predicate<AggregatedHttpResponse> TRUE_PREDICATE = unused -> true;
+
+    BlockingConditionalResponseAs(ResponseAs<HttpResponse, AggregatedHttpResponse> originalResponseAs,
+                                  ResponseAs<AggregatedHttpResponse, ResponseEntity<V>> responseAs,
+                                  Predicate<AggregatedHttpResponse> predicate) {
+        super(originalResponseAs, responseAs, predicate);
+    }
+
+    /**
+     * Adds a mapping such that the response content will be deserialized
+     * to the specified {@link Class} if the {@link Predicate} is satisfied.
+     */
+    public BlockingConditionalResponseAs<V> andThenJson(
```

Comment:
I think we should rename this to `orElseJson`. `andThen` is used when both instances are used. However, if `predicate` returns `false`, then it isn't converted. Let me send a patch to rename these methods.

Reply:
Thanks.
Please send me your suggestion for these method's name.

Reply:
Hi, @my4-dev! I made a few changes:
- Do not block the inside of the converter
  - I realized that this chaining converts also can be used by the `WebClient`. We must not block the inside of the converters.
- Rename `andThenJson` to `orElseJson`
  - `andThen` is used when both instances are applied. e.g. https://github.com/line/armeria/blob/ed16bf6f8846415fdadd70d46afc364376669058/core/src/main/java/com/linecorp/armeria/server/logging/AccessLogWriter.java#L79-L83
  - We apply the converter only when the `Predicate` passes and if it passes the latter one is not applied. So we should rename it to `orElse` instead.
- Remove `ConditionalResponseAs` interface because we don't need it at the moment.

PTAL and let me know if there are changes that I need to revert. 😉